### PR TITLE
Add dupe-ext-id-error to our catchable errors

### PR DESCRIPTION
**Pull request by haakonvt**
https://cognitedata.slack.com/archives/C019TNQJ01E/p1615913546015500

https://cognitedata.slack.com/archives/C010VCGKBCM/p1615926331011700

### CODE REVIEW

File: src/function.py [hunk 1]
```diff
@@ -150,21 +150,23 @@ def upload_and_create(client: CogniteClient, config: FunctionConfig) -> Function
...
         # upsert was requested. delete schedules, function and files
-        try_delete(client=client, external_id=config.external_id)
-
+        try_delete(client, config.external_id)
     try:
         file_id = zip_and_upload_folder(client, config, zip_file_name)
         return create_function_and_wait(client=client, file_id=file_id, config=config)
 
+    except CogniteAPIError as e:
+        if "Function externalId duplicated" in e.message:
```
Comment (pavelzubarev) [inline src/function.py:159]: this is a bit dangerous to rely on the error content.

**Review by pavelzubarev — APPROVED** [review]
looks legit

File: src/function.py (hunk 1 above)
Comment (haakonvt) [reply src/function.py:159]: Yup, but I have no other ideas if we want to keep our `retry` as is 🙁

**Comment by haakonvt** [comment]
New version tested working! Merging 🚀 

Screenshot 2021-03-17 at 09 49 43

**Pull request merged**